### PR TITLE
gitian: Remove Windows 32 bit build

### PR DESCRIPTION
**Pull request by maflcko**
The Windows 32 bit build has been removed from https://bitcoincore.org/en/download/, so unless there are complaints, we don't need to build it even

### CODE REVIEW

File: contrib/gitian-descriptors/gitian-win.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ script: |
   set -e -o pipefail
 
   WRAP_DIR=$HOME/wrapped
-  HOSTS="i686-w64-mingw32 x86_64-w64-mingw32"
+  HOSTS="x86_64-w64-mingw32"
```
Comment (luke-jr) [inline contrib/gitian-descriptors/gitian-win.yml]: This looks so bare. Almost makes me want to build a PPC64 Windows binary just for the sake of it. :)

Comment (MarcoFalke) [reply contrib/gitian-descriptors/gitian-win.yml]: Can you install windows on ppc even?

**Comment by practicalswift** [comment]
Concept ACK

**Comment by laanwj** [comment]
even in the unlikely case that we decide to restore it for 0.18, I think removing it for 0.19 is non-controversial
utACK af04430963cb991f730e83d9a66474b6a93967a

**Comment by achow101** [comment]
utACK faf04430963cb991f730e83d9a66474b6a93967a

File: contrib/gitian-descriptors/gitian-win.yml (hunk 1 above)
Comment (luke-jr) [reply contrib/gitian-descriptors/gitian-win.yml]: NT4 :p

**Comment by practicalswift** [comment]
utACK faf04430963cb991f730e83d9a66474b6a93967a

**Comment by MarcoFalke** [comment]
Removed it from depends and travis as well

File: contrib/gitian-descriptors/gitian-win.yml (hunk 1 above)
Comment (davehamiltone) [reply contrib/gitian-descriptors/gitian-win.yml]: o dear I have been running bitcoin core on a 32-bit windows system for a few years now. Am I the last one? I'd like to upgrade to 0.18 please. Any chance of having 32-bit windows back, or is it too much bother?

**Comment by fanquake** [comment]
Concept ACK

**Comment by laanwj** [comment]
This was bound to be a somewhat controversial decision: however one architecture per platform is great for maintenance-heavy outlier platforms such as Windows (and MacOSX). This allows for more focus in testing and, and hopefully a better user experience and better security, in time. Cross-compilation to windows is fraught with some risks, and they're minimized by only having one toolchain (mingw-w64) to check.

If you need a wider range of architectures it's better to stick with Linux, or one of the BSDs. Another option is to use a VM. I was about to suggest WSL, but [they don't have 32-bit support either](https://superuser.com/questions/1407730/run-32-bit-application-on-ubuntu-on-windows-subsystem-for-linux)!

File: contrib/gitian-descriptors/gitian-win.yml (hunk 1 above)
Comment (luke-jr) [reply contrib/gitian-descriptors/gitian-win.yml]: Why can't you use the 64-bit build?

**Comment by luke-jr** [comment]
I do think we should keep the Travis job around...

**Comment by fanquake** [comment]
There's a few more `win32` references that can be removed:

https://github.com/bitcoin/bitcoin/blame/master/doc/release-process.md#L224

https://github.com/bitcoin/bitcoin/blob/c459c5f70176928adcee4935813a2dbe7f4dbd51/contrib/gitian-build.py#L101

**Comment by MarcoFalke** [comment]
Thanks, done

File: .travis.yml [hunk 2]
```diff
@@ -97,16 +97,6 @@ jobs:
...
-    - stage: test
-      name: 'Win32  [GOAL: deploy]  [no gui or functional tests]'
-      env: >-
-        HOST=i686-w64-mingw32
-        DPKG_ADD_ARCH="i386"
-        PACKAGES="python3 nsis g++-mingw-w64-i686 wine-binfmt wine32"
-        RUN_FUNCTIONAL_TESTS=false
-        GOAL="deploy"
-        BITCOIN_CONFIG="--enable-reduce-exports --disable-gui-tests"
-
```
Comment (luke-jr) [inline .travis.yml]: Let's keep the Travis job...

**Review by luke-jr — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (MarcoFalke) [reply .travis.yml]: Why? I am not aware that this job ever failed and the win64 one didn't. Also, why would we want to waste resources on testing a target that we never ship?
I'd rather have freebsd tests than a win32 one.

Comment (luke-jr) [reply .travis.yml]: We ship source code.

Comment (MarcoFalke) [reply .travis.yml]: It is the responsibility of the user to run the tests. You could argue that many users don't run the tests on their target when they download the gitian binaries, but that doesn't apply here. If some users compile on their own, they need to run the tests themselves.

Comment (luke-jr) [reply .travis.yml]: Travis doesn't exist to run the tests. It exists to help us developers avoid doing things that will break the tests. (Running the tests is just how we accomplish that.)

Comment (MarcoFalke) [reply .travis.yml]: Either we support win32, test and ship it, or we don't. There is no in-between.

Comment (luke-jr) [reply .travis.yml]: I do not agree. We support plenty of things we do not recommend or ship binaries for.

Indeed, the best approach is to compile yourself, which itself falls outside the "ship" scope.

Comment (sdaftuar) [reply .travis.yml]: FWIW I agree with @MarcoFalke here -- seems like not a great use of travis resources to explicitly test a platform that we're no longer interested in continuing to support (who will debug problems if they are found?).  On top of that, reducing the load on travis has benefits; I've noticed that the time I wait from updating a PR to having the travis jobs complete has ticked up recently.  So I'd rather we not spend those travis cycles on win32 unless it was an important platform to test on for some reason, and the decision to drop it strikes me as a statement that it's not.

**Comment by fanquake** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15939/commits/fa193dc8e6f3b96fa2dba2f1c1668f7720fed320

I agree with @MarcoFalke  & @sdaftuar about Travis. I don't think we should have a job dedicated to a binary we aren't shipping.

File: .travis.yml (hunk 2 above)
Comment (luke-jr) [reply .travis.yml]: The decision not to provide binaries isn't necessarily a decision to drop support.

(That being said, *I* certainly am not interested in providing support, so if nobody else is either...)

**Comment by laanwj** [comment]
utACK fa193dc8e6f3b96fa2dba2f1c1668f7720fed320

File: contrib/gitian-descriptors/gitian-win.yml (hunk 1 above)
Comment (jeffrade) [reply contrib/gitian-descriptors/gitian-win.yml]: I'll defer to someone who knows more than me, but can a 64-bit build run on a 32-bit machine?

**Pull request merged**